src/main/reactjs/src/redux/selectors/publicTranslator.ts

```diff
@@ -37,6 +35,26 @@ export const selectedPublicTranslatorsForLanguagePair = createSelector(
 );
 
 // Helpers
+const filterPublicTranslators = (
+  translators: Array<PublicTranslator>,
+  filters: PublicTranslatorFilter
+) => {
+  const isNotEmpty = (v: string) => !Utils.isEmptyString(v);
+  let filteredData = translators;
+  // Filter data only if the criteria are defined
+  if (isNotEmpty(filters.fromLang) && isNotEmpty(filters.toLang)) {
+    filteredData = filteredData.filter((t) => filterByLanguagePair(t, filters));
+  }
```

Comment:
Käyttäjä tulee yllättymään jos hän valitsee vain yhden kieliparin kielistä: silloinhan suodatusta ei tehdä lainkaan.
Jos kieliparia ei kuitenkaan ole pakko syöttää ennen kuin hakee kieliä, olisi myös loogista että erotettaisiin näiden from- ja to-kielten suodatukset toisistaan.

Reply:
Nyt pakotetaan valitsemaan molemmat kielet.